pybird_dev/greenfunction.py

```diff
@@ -1,17 +1,24 @@
+import numpy as np
 from scipy.integrate import quad
 from scipy.special import hyp2f1
+from scipy.integrate import odeint
+
 
 class GreenFunction(object):
-    
-    def __init__(self, Omega0_m, w=None, quintessence=False):
+
+    def __init__(self, Omega0_m, w=None, quintessence=False, Omega_rc=None, MG=False):
```

Comment:
Maybe MG is not needed if Omega_rc sets it?

Reply:
Agreed. The problem is I don't really understand how to modify __init__ for the MG/nDGP cases. My understanding of the current setup is:
1) The default is LCDM with EdS kernels.
2) The user can specify `exact_time' to go beyond EdS kernels for LCDM.
3) The user can specify `wcdm' to go beyond w=-1 -- this automatically activates `exact_time'.
4) The user can specify `quintessence' to activate clustering quintessence -- this automatically activates `wcdm' and therefore also `exact_time'.

Is that correct? What I think the new setup for general scale-independent modified gravity needs to be is:
5) There is an option `MG' which activates the new MG forms for the second-, and third-order coefficients. If `MG' is activated, `exact_time' should automatically activate also.
6) However, the user cannot only specify `MG', as then we don't know what the form for mu(a), mu2(a), mu22(a) should be.
7) The user can specify a specific scale-independent model e.g. `nDGP' or 'jordan-brans-dicke' which sets the behaviour of mu(a), mu2(a), mu22(a) -- this automatically activates `MG' and therefore `exact_time'.
8) The situation gets more complicated if the user needs to be able to choose different forms for H(a) within a specific MG model -- e.g. within `nDGP' the user can choose to compute H(a) with the proper nDGP background, or a wCDM/quintessence behaviour, or fixed to LCDM. This relates to the other comment about line 19.

Do you agree? If so, how do I modify __init__ to yield the above logic?

Reply:
The points 1-4 are correct.
I agree with your logic, which is useful to extend to other models. The philosophy is to add options that trigger others or check if parameters are defined.